From 5ef19d5974684f10ed4cd60191da5b84148da217 Mon Sep 17 00:00:00 2001
From: Thang Tran <thuutran@amperecomputing.com>
Date: Tue, 31 Dec 2024 13:22:10 +0700
Subject: [PATCH 1/2] buttons: support power button whose "name" is
 POWER_BUTTON

After the commit [1], it only supports power buttons that have the
"name" property are POWER_BUTTON1/2/3... It makes old configurations
that support single Host system can not work.

Solution: add one more registy that has "name" is "Power_BUTTON" to allow
to create Dbus object from gpio that has the "name" property is
"POWER_BUTTON".

[1] https://github.com/openbmc/phosphor-buttons/commit/31ce375e7e71ff6b3da460dc63c1ab4f16cf4df6

Upstream-Status: Pending

Tested:
   1. Configure gpio that has "name" property is "POWER_BUTTON"
      in the gpio_defs.json file.
   2. The /xyz/openbmc_project/Chassis/Buttons/Power0 object path
      is created.

Signed-off-by: Thang Tran <thuutran@amperecomputing.com>
---
 src/power_button.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/power_button.cpp b/src/power_button.cpp
index 879c659..08f4829 100644
--- a/src/power_button.cpp
+++ b/src/power_button.cpp
@@ -22,6 +22,9 @@
 static ButtonIFRegister<PowerButton>
     multiButtonRegister(phosphor::button::numberOfChassis());
 
+static ButtonIFRegister<PowerButton>
+   singleButtonRegister;
+
 void PowerButton::simPress()
 {
     pressed();
-- 
2.43.0

